utils/interfaces/_agent.py

```diff
@@ -107,3 +107,17 @@ def add_traces_validation(self, validator, success_by_default=False):
         self.validate(
             validator=validator, success_by_default=success_by_default, path_filters=r"/api/v0\.[1-9]+/traces"
         )
+
+    def get_spans_for_request(self, request):
```

Comment:
could you use the same prototype as in library interface : 

```
def get_spans(self, request=None):
```

* if request is none, iterate over all spans
* otherwise, iterate only on spans triggered by the request.

Tnaks !

Reply:
I can do it yes, although I don't fully agree it's better.

I actually had that version in a previous version. In my opinion, it's a bit confusing right now what actually happens whenever the `request` is passed in a function, and what is being returned in each case. I basically always had to open the code of every function to see what it does, since naming and docs are a bit lacking.

For example, initially, I didn't even notice that the `get_spans` is inside the base `InterfaceValidator`. This was surprising to me since there are 3 places where you can get spans from (tracer->agent, agent->backend, backend->user), so having the agent traces in every interface could lead to bugs. I didn't notice this until the PR CI ran all the tests.

Probably these are issues due to using untyped Python as well, so some better documentation would help in providing clear expectations when calling these helpers.

Reply:
Done.

Reply:
`get_spans` in InterfaceValidator is indeed an error, and I'm almost sure it's not used ^^ I will check and fix that-> #876 (some interface does not have the knowledge of what is a "span", so there is no point to have it here).

For namings, totally open to suggestion, as long as the code is coherent. I'll at least add some documentations